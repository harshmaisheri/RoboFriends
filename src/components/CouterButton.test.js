import React from "react";
import { shallow } from "enzyme";
import CounterButton from "./CounterButton";

const mockColor = "red";
const Wrapper = shallow(<CounterButton color={mockColor} />);

it("Expect to Render Card List", (done) => {
  expect(Wrapper.debug()).toMatchSnapshot();
  done();
});

it("Correctly increments the counter number", (done) => {
  Wrapper.find('[id="counter"]').simulate("click");
  expect(Wrapper.state()).toEqual({ count: 1 });
  Wrapper.find('[id="counter"]').simulate("click");
  expect(Wrapper.state()).toEqual({ count: 2 });
  done();
});
