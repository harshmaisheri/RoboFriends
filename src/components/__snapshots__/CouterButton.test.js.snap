// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to Render Card List 1`] = `
"<button id=\\"counter\\" color=\\"red\\" onClick={[Function]}>
  Count: 
  0
</button>"
`;
